somaticseq/somatic_vcf2tsv.py

```diff
@@ -0,0 +1,1390 @@
+            
+            ##### ##### ##### ##### ##### #####
+            for my_coordinate in my_coordinates:
+                            
+                ######## If VCF, can get ref base, variant base, as well as other identifying information ######## 
+                if is_vcf:
+                    
+                    ref_bases = []
+                    alt_bases = []
+                    indel_lengths = []
+                    all_my_identifiers = []
+                    
+                    for variant_i in variants_at_my_coordinate:
+    
+                        ref_base = variant_i.refbase
+                        first_alt = variant_i.altbase.split(',')[0]
+                        indel_length = len(first_alt) - len(ref_base)
+    
+                        ref_bases.append( ref_base )
+                        alt_bases.append( first_alt )
+                        indel_lengths.append( indel_length )
+                        
+                        # Extract these information if they exist in the VCF file, but they could be re-written if dbSNP/COSMIC are supplied.
+                        if_dbsnp  = 1 if re.search(r'rs[0-9]+', variant_i.identifier) else 0
+                        if_cosmic = 1 if re.search(r'COS[MN][0-9]+', variant_i.identifier) else 0
+                        if_common = 1 if variant_i.get_info_value('COMMON') == '1' else 0
+                        num_cases = variant_i.get_info_value('CNT') if variant_i.get_info_value('CNT') else nan
+                        
+                        if variant_i.identifier == '.':
+                            my_identifier_i = set()
+                        else:
+                            my_identifier_i = variant_i.identifier.split(';')
+                            my_identifier_i = set( my_identifier_i )
+                        
+                        all_my_identifiers.append( my_identifier_i )
+                                    
+                ## If not, 1) get ref_base, first_alt from other VCF files. 
+                #          2) Create placeholders for dbSNP and COSMIC that can be overwritten with dbSNP/COSMIC VCF files (if provided)
+                else:
+                    variants_at_my_coordinate = [None] # Just to have something to iterate
+                    ref_base = first_alt = indel_length = None
+                    
+                    # Could be re-written if dbSNP/COSMIC are supplied. If not, they will remain NaN.
+                    if_dbsnp = if_cosmic = if_common = num_cases = nan
+    
+                # Keep track of NumCallers:
+                num_callers = 0
+                
+                #################################### Find the same coordinate in those VCF files ####################################
+                if mutect:  got_mutect,  mutect_variants,  mutect_line  = genome.find_vcf_at_coordinate(my_coordinate, mutect_line,  mutect,  chrom_seq)
```

Comment:
The following series of `if` statements can easily be refactored to a small for-loop.